tools/walletextension/keymanager/keymanager.go

```diff
@@ -0,0 +1,287 @@
+		logger.Info("found sealed encryption key")
+		return encryptionKey, nil
+	}
+
+	// TODO: Handle the case where we have to exchange the key with another enclave
+
+	logger.Info("key exchange url", "keyExchangeURL", config.KeyExchangeURL)
+
+	if config.KeyExchangeURL != "" {
+		logger.Info("Inside Key exchange process")
+		privateKey, publicKeyBytes, err := GenerateECDSAKeys()
+		if err != nil {
+			logger.Crit("unable to generate ECDSA keys", log.ErrKey, err)
+			return nil, err
+		}
+
+		logger.Info("exchanging key with another enclave", "keyExchangeURL", config.KeyExchangeURL)
+		pubKeyHash := sha256.Sum256(publicKeyBytes)
+		attestationReport, err := GetReport(pubKeyHash[:])
+		if err != nil {
+			logger.Crit("unable to get attestation report", log.ErrKey, err)
+			return nil, err
+		}
+		logger.Info("Successfully received attestation report")
+
+		// Serialize the attestation report
+		reqBody, err := json.Marshal(attestationReport)
+		if err != nil {
+			logger.Crit("unable to marshal attestation report", log.ErrKey, err)
+			return nil, err
+		}
+
+		// Send the request to the other enclave
+		resp, err := http.Post(config.KeyExchangeURL+"/v1"+common.PathKeyExchange, "application/json", bytes.NewBuffer(reqBody))
+		if err != nil {
+			logger.Crit("unable to send attestation report", log.ErrKey, err)
+			return nil, err
+		}
+		defer resp.Body.Close()
+
+		// print the response status code
+		logger.Info("response status code", "status", resp.StatusCode)
+		logger.Info("response headers", "headers", resp.Header)
+		logger.Info("response body", "body", resp.Body)
+
+		// Check the response status code
+		if resp.StatusCode != http.StatusOK {
+			logger.Crit("received non-200 status code", "status", resp.StatusCode)
+			return nil, fmt.Errorf("received non-200 status code: %d", resp.StatusCode)
+		}
+
+		// Read the response body
+		responseBody, err := io.ReadAll(resp.Body)
+		if err != nil {
+			logger.Crit("unable to read response body", "error", err)
+			return nil, err
+		}
+
+		// Parse the JSON response - this will fail if the response isn't valid JSON
+		var responseMap map[string]string
+		if err := json.Unmarshal(responseBody, &responseMap); err != nil {
+			logger.Crit("unable to unmarshal response", "error", err)
+			return nil, err
+		}
+
+		// Get the encrypted key from the response
+		encryptedKeyEncoded, ok := responseMap["encryptedKey"]
+		if !ok {
+			err := fmt.Errorf("encryptedKey not found in response")
+			logger.Crit("error parsing response", "error", err)
+			return nil, err
+		}
+
+		// Decode and decrypt the encrypted key
+		decryptedKey, err := DecodeAndDecrypt(encryptedKeyEncoded, privateKey)
+		if err != nil {
+			logger.Crit("unable to decode and decrypt the encrypted key", "error", err)
+			return nil, err
+		}
+
+		// Now you have the encryption key
+		logger.Info("Successfully retrieved and decrypted the encryption key")
+
+		fmt.Println("decrypted key: ", decryptedKey)
+
+		return decryptedKey, nil
```

Comment:
shouldn't we seal/persisting key in this case too before we return?

Reply:
fixed